client/python/examples/README.md

```diff
@@ -0,0 +1,185 @@
+
+```bash
+export ARMADA_SERVER=localhost
+export ARMADA_PORT=443
+export DISABLE_SSL=true
+```
+
+You can then simply run one of the examples
+
+```bash
+python3 simple.py
+```
+
+## Example Files
+
+### simple.py
+
+Example of using the armada client to create a queue, jobset and job,
+then watch for the job to succeed or fail.
+
+#### Walkthrough
+
+```py
+def create_dummy_job(client: ArmadaClient):
+```
+Armada uses kubernetes to control jobs, so we need to define a podspec that are job will use.
+
+```py
+if DISABLE_SSL:
+    channel = grpc.insecure_channel(f"{HOST}:{PORT}")
+else:
+    channel_credentials = grpc.ssl_channel_credentials()
+    channel = grpc.secure_channel(
+        f"{HOST}:{PORT}",
+        channel_credentials,
+    )
+```
+
+If you are using a remote server, secure_channel should be used to ensure that all infomation is encrypted.
+
+
+```py
+client.create_queue(name=queue, priority_factor=1)
+```
+
+All queues need a priority factor to be created, which is used to determine the order in which jobs are run.
+
+### general.py
+
+A more fledged out version of `simple.py` where we create a queue only if it doesn't exist, and then create a jobset and job, and wait until the job succeeds or fails.
+
+#### Walkthrough
+
+```py
+def wait_for_job_event(client, event_stream, job_id: str, event_state: EventType)
+```
+
+We use this function to check the status of a job, return false if the job failed and otherwise wait for the event state we are interested in.
+
+```py
+job_id = resp.job_response_items[0].job_id
+```
+
+We can use this code to get the job_id we are interested in from the event_stream.
+
+
+```py
+client.reprioritize_jobs(new_priority=2, queue=queue, job_set_id=job_set_id)
+
+# Needed to allow for the delay in the job_set being created
+time.sleep(2)
+```
+
+As the code documents, we need the sleep to wait for the job_set to be created, allowing for `client.get_job_events_stream` to not fail.
+
+```py
+try:
+    client.create_queue(name=queue, priority_factor=1)
+
+# Handle the error we expect to maybe occur
+except grpc.RpcError as e:
+    code = e.code()
+    if code == grpc.StatusCode.ALREADY_EXISTS:
+        print(f"Queue {queue} already exists")
+        client.update_queue(name=queue, priority_factor=1)
+    else:
+        raise e
+```
+
+For any grpc-related errors, more detail can be found by checking the grpc.RpcError object. In this case, we use it to ignore if a queue has already been created or not.
+
+### async_logging.py
+
+Demonstrates how to run jobs, and also log all changes to that job or job_set in realtime concurrently.
+
+#### Walkthrough
+
+```py
+def watch_job_set(client: ArmadaClient, queue: str, job_set_id):
+```
+
+Watches all changes on a job_set. It attempts to connect multiple times instead of sleeping like in `general.py`. It also demonstrates using the clients `unmarshal_event_response()` method to better access the event object.
+
+```py
+def watch_queue(client, queue):
```

Comment:
We should ask if this would be useful before we put in examples?

Reply:
Maybe discuss this on Wednesday?

Reply:
Sure but I don't know if any of us have a good answer for you.  I'd ask the GR Armada team about this.

Reply:
I talked with @d80tb7 and I think we should remove watch_queue everywhere.

Reply:
Okay. I think its just used in `async_logging` so I will remove that shortly